arm/proofs/decode.ml

```diff
@@ -296,11 +296,11 @@ let decode = new_definition `!w:int32. decode w =
     else if q then
       let pos = (val imm4) * 8 in
       // datasize is fixed to 128.
-      SOME (arm_EXT (QREG' Rd) (QREG' Rm) (QREG' Rn) pos)
+      SOME (arm_EXT (QREG' Rd) (QREG' Rn) (QREG' Rm) pos)
```

Comment:
The order of `Rm` and `Rn` has been swapped. In the end this wasn't the source of a bug because `arm_EXT`'s arguments was also in swapped order (`Rm` and `Rn`). I fixed this and the arguments of `arm_EXT`. To check that there isn't any mistake, the running simulator is also testing EXT.